joomla/joomla-cms
# [4.1] Smart Search: Fix missing taxonomy language string

### Summary of Changes
Smart Search tries to create its own language keys for the indexed taxonomies. Unfortunately, when there is no language string for the key, we get a "Search by PLG_FINDER_QUERY_FILTER_BRANCH_S_SOMETHING" in the search form in the frontend. This PR checks if a translation for the language key exists and if there is none (and debug is not enabled) it will instead display the taxonomy title instead.

### Testing Instructions
1. If you don't have the effect already, we need to add a fake taxonomy. Go to plugins/finder/content/content.php and add `$item->addTaxonomy('Test', 'Strange');` after line 342.
2. Reindex your site with the button in the backend.
3. Go to the advanced search form in the frontend and see a dropdown with the label "Search by PLG_FINDER_QUERY_FILTER_BRANCH_S_TEST".
4. Enable debugging and see that the string didn't change.
5. Apply PR.
6. See that the string didn't change with debugging enabled.
7. Disable debugging.
8. See now that the string is "Search by Test"
